ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs

```diff
@@ -25,27 +28,70 @@ import           Test.Ouroboros.Storage.ChainDB.StateMachine (AllComponents,
 import           Test.Ouroboros.Storage.TestBlock
 import           Test.Tasty (TestTree, testGroup)
 import           Test.Tasty.HUnit (Assertion, assertFailure, testCase)
+import           Test.Util.Tracer (recordingTracerTVar)
 
+import           Ouroboros.Consensus.Ledger.SupportsProtocol
+                     (LedgerSupportsProtocol)
 import           Ouroboros.Consensus.Util.ResourceRegistry (closeRegistry,
                      unsafeNewRegistry)
 import           Test.Util.ChainDB (MinimalChainDbArgs (..), emptyNodeDBs,
                      fromMinimalChainDbArgs, nodeDBsVol)
 
-import           Ouroboros.Consensus.Config (TopLevelConfig)
+import           Ouroboros.Consensus.Config (TopLevelConfig,
+                     configSecurityParam)
 import           Ouroboros.Consensus.Ledger.Extended (ExtLedgerState)
+import           Ouroboros.Consensus.Storage.ChainDB
+                     (Internal (intCopyToImmutableDB))
 import           Ouroboros.Consensus.Storage.ImmutableDB.Chunks as ImmutableDB
 
+
 tests :: TestTree
-tests = testGroup "First follower instruction isJust on empty ChainDB"
-  [ testCase "model" $ runModel' getFollowerInstruction
-  , testCase "system" $ runSystem' getFollowerInstruction
+tests = testGroup "Unit tests"
+  [ testGroup "First follower instruction isJust on empty ChainDB"
+    [ testCase "model" $ runModel' getFollowerInstruction
+    , testCase "system" $ runSystem' getFollowerInstruction
+    ]
+  , testGroup "ouroboros-network-4183"
+    [ testCase "model" $ runModel' ouroboros_network_4183
+    , testCase "system" $ runSystem' ouroboros_network_4183
+    ]
   ]
 
+
+getFollowerInstruction :: (SupportsUnitTest m, MonadError TestFailure m) => m ()
+getFollowerInstruction = do
+  f <- newFollower
+  followerInstruction f >>= \case
```

Comment:
In the specific situation of pattern matching on the result of a monadic action I prefer it this way, to avoid having to introduce a superfluous name.